core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -202,7 +202,9 @@ class ReplicaStateMachine(config: KafkaConfig,
           controllerBrokerRequestBatch.addStopReplicaRequestForBrokers(Seq(replicaId), replica.topicPartition,
             deletePartition = false, (_, _) => ())
         }
-        val replicasToRemoveFromIsr = validReplicas.filter(replica => controllerContext.partitionLeadershipInfo.contains(replica.topicPartition))
+        val (replicasToRemoveFromIsr, replicasWithoutLeadershipInfo) = validReplicas.partition { replica =>
```

Comment:
To be consistent, replicasToRemoveFromIsr probably should be named replicasWithLeadershipInfo?

Reply:
Good point, I'll fix that.